trinity/utils/filesystem.py

```diff
@@ -0,0 +1,22 @@
+import os
+
+
+def is_same_path(p1, p2):
+    n_p1 = os.path.abspath(os.path.expanduser(p1))
+    n_p2 = os.path.abspath(os.path.expanduser(p2))
+
+    try:
+        return os.path.samefile(n_p1, n_p2)
+    except FileNotFoundError:
+        return n_p1 == n_p2
+
+
+def is_under_path(base_path, path, strict=True):
```

Comment:
`base_path` was a confusing name for me here. I thought it meant the opposite, on the first read. Maybe part of the reason is that I was expecting arguments in reverse order: `(parent_path, maybe_child_path, strict=True)`

How will this get used eventually? Now that you've got me looking for currying everywhere, I imagine the more curry-friendly version would be reverse the order of these args. Like `is_in_data_dir = is_under_path(data_dir)`.